backend/src/requests/user/requests-user.service.ts

```diff
@@ -1,19 +1,93 @@
 import {
+  HttpException,
+  HttpStatus,
   Injectable,
   InternalServerErrorException,
+  NotAcceptableException,
+  NotFoundException,
   UnauthorizedException,
 } from '@nestjs/common';
 import { LoginAttempt } from '@prisma/client';
+import { ExpertService } from '../../database/expert/expert.service';
 import { LoginAttemptService } from '../../database/login-attempt/login-attempt.service';
 import { UserService } from '../../database/user/user.service';
+import { UpdateUser } from './dto/update-user.dto';
+
+enum Registration { // TODO extract
+  registered = 'registered',
+  proUser = 'proUser',
+}
 
 @Injectable()
 export class RequestsUserService {
   constructor(
     private readonly userLoginAttemptService: LoginAttemptService,
     private readonly userService: UserService,
+    private readonly expertService: ExpertService,
   ) {}
 
+  /**
+   * Get information on a user identified by its id
+   * @returns user if found, object after the defined schema in OpenAPI
+   * @throws NotFoundException if no user found with id
+   * */
+  async getProfileWrapper(userId: string) {
+    const usrProfile = await this.userService.getUser(userId);
+    const expertTags = await this.expertService.getExpertTagsForUser(userId);
+
+    if (usrProfile === null || expertTags === null) {
+      throw new NotFoundException('User does not exist');
+    }
+    const accountState: Registration = usrProfile.isPro
+      ? Registration.proUser
+      : Registration.registered;
+    const expertTagList: string[] = expertTags.map((e) => e.tagname);
+
+    return {
+      name: usrProfile.username,
+      profilePicture: null,
+      registrationDate: usrProfile.timeOfRegistration,
+      accountState: accountState.toString(),
+      expertTopics: expertTagList,
+      userStatistics: {
+        TODO: null,
+      },
+      activityPoints: usrProfile.activityPoints,
+    };
+  }
+
+  /**
+   * Update the user (currently only the name)
+   * @param req the requests data
+   * @param data the body of the request including the payload to update the user
+   * @return the updated user
+   * @throws NotFoundException if the user does not exists
+   * @throws NotModified if nothing changes
+   * @throws NotAcceptableException if the payload is invalid
+   * */
+  async updateUser(req: any, data: UpdateUser) {
+    const user = await this.userService.getUser(req.userId);
+    if (user == null) throw new NotFoundException('No user found with this id');
+
+    if (user.username == data.name || Object.keys(data).length == 0)
+      throw new HttpException(
+        'Nothing will be changed',
+        HttpStatus.NOT_MODIFIED,
+      );
+
+    if (!data) {
```

Comment:
If data is nullable, shouldn't it be checked in before the other method calls?
I mean, it's checked `data.name` before this check